tests/test_asd.py

```diff
@@ -30,23 +30,23 @@ def test_validate_asd(caplog):
 clampon_validation = [(0, 50, 0),
                       (20, 70, 14500),
                       (50, 90, 38500),
-                      (-10, 400, None),
+                      (-10, 400, np.nan),
                       (22, 400, 38196),
                       (70, 500, 198900)]
 @pytest.mark.parametrize('v_mix, GLR, E', clampon_validation)
 def test_clampon(v_mix, GLR, E):
-    assert std_step_clampon(v_mix, GLR) == E
+    np.testing.assert_equal(std_step_clampon(v_mix, GLR), E)
```

Comment:
Jeg foretrekker at vi heler gjør det som vi hadde det og tar vekk den testen. Vi tester valideringen uansett i test_validate_asd funksjonen så det er ikke vits å gjøre det her igjen. Da vil vi også gjøre alle assertions like (konsistens er bra)

Reply:
Enig i at konsistens er bra, men dersom vi fjerner linjen hvor vi tester at np.nan returneres i step-funksjonene har vi ingen tester som verifiserer at np.nan faktisk returneres når validate_asd returnerer True. I validate_asd kan vi kun teste at True returneres for negativ input og ikke at step-funksjonene korrekt returnerer np.nan. Vi kan naturligvis skrive en ekstra testfunksjon som tester at np.nan returneres korrekt

Reply:
Godt poeng.